### PR TITLE
Mark reactivex packages as optional in OSGi metadata

### PR DESCRIPTION
**Pull request by sauthieg**
When trying the `3.5.0.Beta1` bundle, it was unable to deploy when rxjava2 was missing.

I've added `resolution:=optional` to theses new imported packages (just like what was done for other languages).

I've also discovered that `io.vertx.ext.auth` was actually never used in this module, but import was required. I've simply removed the lines related to this group of packages.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -193,15 +193,17 @@
             org.codehaus.groovy.*;resolution:=optional,\
             kotlin.*;resolution:=optional,\
             io.vertx.groovy.*;resolution:=optional,\
-            io.vertx.ext.auth,\
-            io.vertx.ext.auth.*;resolution:=optional,\
```
Comment (cescoffier) [inline pom.xml]: Why are these removed and not io.vertx.rxjava.ext.auth ?

File: pom.xml [hunk 2]
```diff
@@ -193,15 +193,17 @@
...
             io.vertx.lang.rxjava.*;resolution:=optional,\
+            io.vertx.lang.reactivex.*;resolution:=optional,\
             io.vertx.lang.groovy.*;resolution:=optional,\
             io.vertx.codegen.annotations;resolution:=optional,\
             io.vertx.rx.java;resolution:=optional,\
             io.vertx.rxjava.core.*;resolution:=optional,\
             io.vertx.rxjava.ext.auth.*;resolution:=optional,\
+            io.vertx.reactivex;resolution:=optional,\
```
Comment (cescoffier) [inline pom.xml]: Cant' we merge both instructions into one:

io.vertx.reactivex.*;resolution:=optional.

Is there other required packages that must be mandatory ?

File: pom.xml (hunk 1 above)
Comment (sauthieg) [reply pom.xml]: I forgot them :)
thanks for the head-up

File: pom.xml (hunk 2 above)
Comment (sauthieg) [reply pom.xml]: Well, there are MQTT classes generated in `io.vertx.reactivex.mqtt`, that seems weird to optionally import something that is already inside of the bundle

Comment (cescoffier) [reply pom.xml]: Is there a dependency on auth ? Some module relies on vertx-auth to handle the authentication.

**Comment by ppatierno** [comment]
@cescoffier I'm not an OSGi expert so ... can I merge this PR after your change request now ? :)

**Comment by cescoffier** [comment]
@ppatierno +1, go ahead. And thanks @sauthieg

File: pom.xml (hunk 2 above)
Comment (sauthieg) [reply pom.xml]: No code in this bundle use auth, It should never have been there I think.
My bad, in my initial contribution, I forgot to remove them ..
